controllers/dashboard.py

```diff
@@ -36,11 +85,10 @@ def index():
 
     course = db(db.courses.id == auth.user.course_id).select().first()
     assignments = db(db.assignments.course == course.id).select(db.assignments.ALL, orderby=db.assignments.name)
+    chapters = db(db.chapters.course_id == course.base_course).select(orderby=db.chapters.chapter_num)
+
     logger.debug("getting chapters for {}".format(auth.user.course_name))
-    chapters = db(db.chapters.course_id == course.base_course).select()
-    chap_map = {}
-    for chapter in chapters:
-        chap_map[chapter.chapter_label] = chapter.chapter_name
+    get = Get(chapters) #yes this is a dumb name
```

Comment:
You could add a map to the sub_chapter numbers here with a query like
```
sub_chapters = db(db.sub_chapters.chapter_id == chapter.id).select()
for subc in sub_chapters:
   # build a dictionary of dictionaries where the outer index is the chapter and the inner maps the subchapter label to number?
```

Reply:
But that's what I do in Lines 40-44, in the Get class definition?

Reply:
OK, you said you had a bug in that you couldn't use the correct number for exercises and other subchapters that appear in multiple chapters.  But that does give you the right info to use the correct number.

Reply:
The latest commit fixed that bug.